### PR TITLE
Report's implementation section

### LINKED ISSUE #24: Report - Implementation Section
Use outline as a starting point, feel free to modify it as you see fit.

### PR DESCRIPTION
**Pull request by diegocasmo**
Closes #24

### CODE REVIEW

File: report/sections/implementation.tex [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+\subsection{kNN}
 
-\todo{TODO: What is a SVM (short intro)?, hyper-parameters setup and what each of them mean}
+A kNN classifier simply classifies samples by a majority vote of its neighbors, with a sample being assigned to the class most common among its \texttt{k} nearest neighbors \cite{data-mining-intro}. The hyper-parameter \texttt{n\_neighbors} determines the number of neighbors to use to choose the label of a sample, a total of 15 neighbors attained the best results.
 
-\subsection{Multilayer Perceptron (MLP)}
+\subsection{SVM}
 
-\todo{TODO: What is a MLP (short intro)?, hyper-parameters setup and what each of them mean}
+A SVM is a supervised learning algorithm which creates a hyperplane that represents the input samples as points in space, such that samples of different classes are separated from each other. New samples are then mapped into that same space and predicted to belong to a class based on where in that space these are located \cite{data-mining-intro}.
```
Comment (tristaaan) [inline report/sections/implementation.tex:20]: I'm assuming we're spelling these out before this section?

File: report/sections/implementation.tex [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
+
+\subsection{MLP}
+
+A MLP is a feed-forward artificial neural network that uses supervised learning with backpropagation for training the network (update its weights) to essentially map a weighted input to a label. The MLP uses the \texttt{adam} solver, which is a stochastic gradient-based optimizer, a logistic sigmoid function as its activation function, a single hidden layer with a total of 5 neurons, and a maximum of 2000 iterations to reach convergence.
```
Comment (tristaaan) [inline report/sections/implementation.tex]: -"...gradient-based optimizer **[semicolon]**..." optional, I'll amend later

**Review by tristaaan — APPROVED** [review]
One microscopic change.

File: report/sections/implementation.tex (hunk 1 above)
Comment (diegocasmo) [reply report/sections/implementation.tex:20]: Yup, in the intro section

**Pull request merged**